src/unicode/mod.rs

```diff
@@ -33,6 +33,49 @@ impl<S1: AsRef<str>, S2: AsRef<str>> PartialEq<Unicode<S2>> for Unicode<S1> {
     }
 }
 
+impl<S1: AsRef<str>> Unicode<S1> {
+    /// Returns true if the given pattern matches a sub-slice of this string slice.
+    ///
+    /// Returns false if it does not.
+    #[inline]
+    pub fn contains<S2: AsRef<str>>(&self, pat: &Unicode<S2>) -> bool {
+        let mut left = self.0.as_ref().chars().flat_map(lookup);
+        let mut pat = pat.0.as_ref().chars().flat_map(lookup);
+
+        match pat.next() {
+            Some(p0) => 'out: loop {
+                match left.next() {
+                    Some(e) if e == p0 => {
+                        let mut left = left.clone();
+                        let mut pat = pat.clone();
+
+                        loop {
+                            let p = match pat.next() {
+                                None => break 'out true,
```

Comment:
All these breaks could just be returns right? I think that would be simpler

Reply:
In fact, depending on if there's a minimal supported rust version, you could even write
```rust
let Some(p) = pat.next() else { return true };
```